### DISCUSSION TITLE
how can i allocate user network traffic on strongswan?

### DISCUSSION THREAD

**Discussion by kulame** [strongswan/strongswan#1144] [General]
hi all

I have created a strongswan service and want to give each user a 100gb limit per month. how can i do it on strongswan?

**Comment by strongX509** [comment]
RADIUS Accounting can measure the number of bytes a user is transferring through an IPsec tunnel, see

https://docs.strongswan.org/docs/5.9/plugins/eap-radius.html#_accounting

for details. A special [updown](https://docs.strongswan.org/docs/5.9/plugins/updown.html) script could then accumulate all the RADIUS Accounting records for a given user (PLUTO_PEER_ID) and **not** insert an **iptables** pass rule on the VPN gateway if the monthly quota has been reached.

**Reply by tobiasbrunner** [reply]
Authentication via RADIUS could perhaps also enforce this. And with [DAE](https://docs.strongswan.org/docs/5.9/plugins/eap-radius.html#_session_timeout_and_dynamic_authorization_extension) the RADIUS server could also terminate sessions at runtime.
